libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/FileInterceptor.java

```diff
@@ -71,6 +71,11 @@ public static void intercept(@Advice.AllArguments Object[] args, @Advice.Origin
         boolean isMutating = name.equals("move") || name.equals("write") || name.startsWith("create");
         final boolean isDelete = isMutating == false ? name.startsWith("delete") : false;
 
+        // This is Windows implementation of UNIX Domain Sockets (close)
+        if (walker.getCallerClass().getName().equalsIgnoreCase("sun.nio.ch.PipeImpl$Initializer$LoopbackConnector") == true) {
```

Comment:
On Windows, the UNIX domain sockets are implemented as file based operations, however there is no way to say of the file path is domain socket or not (the private APIs are there but not public one) :(( Checking the caller is the only reliable way that I found out for now.

Reply:
TY @reta , I ran into the problem on https://github.com/opensearch-project/job-scheduler/actions/runs/14370974807/job/40293845139?pr=759 and see `sun.nio.ch.PipeImpl$Initializer$LoopbackConnector` in the call stack.

Reply:
Yep, thanks! I also added NetPermission check instead of just trusting the code, thanks @cwperks !